### PR TITLE
add a setting for exponential view distance adjustment

### PR DESCRIPTION
**Pull request by DS-Minetest**
similar to #3404 but disabled by default
this is my first pull request and my first real action in github
sorry for my bad English :cat2: :rainbow:

### CODE REVIEW

**Comment by DS-Minetest** [comment]
:cat2:

**Comment by PilzAdam** [comment]
What is the benefit of using a factor instead of a linear step here?

**Comment by DS-Minetest** [comment]
1. if you want for example make a screenshot but also want a bit fog, its faster to set the view range
2. if you handle with far view range, you don't want to continue with only 10 nodes per press

**Comment by paramat** [comment]
:-1: I often want to set view range to a precise high number, like 1000 etc. Adjustment is fast if you look at the sky or ground.

**Comment by DS-Minetest** [comment]
1. the setting is usually off for such a sort of player, so it can't hurt
2. it takes more than 3 seconds to turn it down to 20 without lag, when lag and not lag changes fast, you have to adjust it very often
3. your mouse gets very slow at lag and it takes much time to look down

**Comment by PilzAdam** [comment]
@DS-Minetest the viewing range is not something that you are constantly adjusting when you play normally. You set it to a reasonable value once and then leave it like this for a long time.

**Comment by paramat** [comment]
Oh sorry i just saw it was optional.

**Comment by DS-Minetest** [comment]
@PilzAdam I know.

**Comment by kwolekr** [comment]
FWIW, the feature is well coded so I don't have any complaints there.

Is this feature actually needed though?  Probably not, it seems a bit superfluous to me.  I'm not +1 or -1ing because I don't have a very strong opinion on the matter.

**Comment by HybridDog** [comment]
>Is this feature actually needed though?

automatic vrange adjustment recently got removed, so l often need to change vrange manually

**Comment by celeron55** [comment]
I think the main issue here is the amount of added code that practically nobody will ever use, but which still has to be maintained in working order by those who don't use it.

Maybe the adjustment could be just switched to always be multiplication-based instead of increasing maintenance work by adding the unnecessary complexity of this bunch of conditionals?

**Comment by HybridDog** [comment]
>Maybe the adjustment could be just switched to always be multiplication-based instead of increasing maintenance work by adding the unnecessary complexity of this bunch of conditionals?

There are people who want linear vrange increasing.
https://github.com/minetest/minetest/pull/3404#issuecomment-160193291

**Comment by paramat** [comment]
:-1: I don't feel it's needed enough, situations where FPS is extrmeely low are rare and there is a workaround (looking at the ground/sky).
I do find often i need to set a far view range precisely with +-10 precision.

**Comment by kwolekr** [comment]
Hmm, thinking about it a little more, the use case here is so limited that it doesn't make sense to add all that code for that purpose.  It's overcomplicating something that really does not need to be overcomplicated.

I would give it a +1 for code review but a -1 for concept approval.  So, overall :-1: for me.  Sorry :/

File: src/game.cpp [hunk 1]
```diff
@@ -3091,8 +3091,21 @@ void Game::toggleProfiler(float *statustext_time, u32 *profiler_current_page,
...
-	s16 range_new = range + 10;
+	static const float factor = g_settings->getFloat("vrange_adjustment_factor");
+	static const bool multiply_range = factor > 1.0f;
+
+	const s16 range = g_settings->getS16("viewing_range");
+
+	s16 range_new;
+
+	if (multiply_range) {
+		range_new = ceil(range * factor);
```
Comment (SmallJoker) [inline src/game.cpp]: `range_new = MYMAX(ceil(range * factor), 20);` shortens it to one line

**Comment by kwolekr** [comment]
Closing this PR due to amount of disapproval...

**Pull request closed without merging**